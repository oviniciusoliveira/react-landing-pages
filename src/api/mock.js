export const twoSectionColumnsMock = {
  __component: 'section.section-two-columns',
  title: 'any_title',
  description: 'any_text',
  metadata: {
    background: true,
    section_id: 'any_section_id',
  },
  __v: 1,
  image: {
    url: 'any_image_src.svg',
  },
};

export const mapSectionContentMock = {
  __component: 'section.section-content',
  title: 'any_title',
  content: '<p>any_content</p>',
  metadata: {
    background: true,
    section_id: 'any_section_id',
  },
};

export const mapSectionGridTextMock = {
  __component: 'section.section-grid',
  title: 'any_section_title',
  description: 'any_section_description',
  text_grid: [
    {
      title: 'any_text_grid_title',
      description: 'any_text_grid_description',
      id: 'any_text_grid_id',
    },
  ],
  image_grid: [],
  metadata: {
    background: true,
    section_id: 'any_section_id',
  },
};

export const mapSectionGridImageMock = {
  __component: 'section.section-grid',
  title: 'any_section_title',
  description: 'any_section_description',
  text_grid: [],
  image_grid: [
    {
      image: {
        name: 'any_image_name',
        alternativeText: 'any_image_alternativeText',
        url: 'any_image_url',
      },
      id: 'any_image_grid_id',
    },
  ],
  metadata: {
    background: true,
    section_id: 'any_section_id',
  },
};
