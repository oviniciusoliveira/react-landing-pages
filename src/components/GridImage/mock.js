export default {
  withBackground: false,
  title: 'My grid',
  sectionId: 'grid-image',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus malesuada arcu sed dictum malesuada.',
  grid: [
    {
      id: '1',
      alternativeText: 'Teste 1',
      imageSrc: 'https://source.unsplash.com/random/800x800?r=1',
    },
    {
      id: '2',
      alternativeText: 'Teste 2',
      imageSrc: 'https://source.unsplash.com/random/800x800?r=2',
    },
    {
      id: '3',
      alternativeText: 'Teste 3',
      imageSrc: 'https://source.unsplash.com/random/800x800?r=3',
    },
    {
      id: '4',
      alternativeText: 'Teste 4',
      imageSrc: 'https://source.unsplash.com/random/800x800?r=4',
    },
    {
      id: '5',
      alternativeText: 'Teste 5',
      imageSrc: 'https://source.unsplash.com/random/800x800?r=5',
    },
    {
      id: '6',
      alternativeText: 'Teste 6',
      imageSrc: 'https://source.unsplash.com/random/800x800?r=6',
    },
  ],
};
