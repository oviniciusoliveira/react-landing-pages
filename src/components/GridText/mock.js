export default {
  withBackground: false,
  title: 'My grid',
  sectionId: 'grid-text',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus malesuada arcu sed dictum malesuada.',
  grid: [
    {
      id: '1',
      title: 'Teste 1',
      description:
        'Vivamus vitae porta diam. Nullam mauris lorem, pretium ut dapibus sed, efficitur ac lorem. Phasellus eget felis leo. Integer vel suscipit nulla. Integer eget gravida lorem. Fusce odio nibh, pellentesque nec ultricies eu, molestie non purus. Cras condimentum porta pharetra. Vivamus porta laoreet leo in iaculis. Donec eget laoreet risus. Etiam sed erat ultrices, sodales sem sit amet, sagittis lacus. Pellentesque in cursus enim. ',
    },
    {
      id: '2',
      title: 'Teste 2',
      description:
        'Vivamus vitae porta diam. Nullam mauris lorem, pretium ut dapibus sed, efficitur ac lorem. Phasellus eget felis leo. Integer vel suscipit nulla. Integer eget gravida lorem. Fusce odio nibh, pellentesque nec ultricies eu, molestie non purus. Cras condimentum porta pharetra. Vivamus porta laoreet leo in iaculis. Donec eget laoreet risus. Etiam sed erat ultrices, sodales sem sit amet, sagittis lacus. Pellentesque in cursus enim. ',
    },
    {
      id: '3',
      title: 'Teste 3',
      description:
        'Vivamus vitae porta diam. Nullam mauris lorem, pretium ut dapibus sed, efficitur ac lorem. Phasellus eget felis leo. Integer vel suscipit nulla. Integer eget gravida lorem. Fusce odio nibh, pellentesque nec ultricies eu, molestie non purus. Cras condimentum porta pharetra. Vivamus porta laoreet leo in iaculis. Donec eget laoreet risus. Etiam sed erat ultrices, sodales sem sit amet, sagittis lacus. Pellentesque in cursus enim. ',
    },
  ],
};
