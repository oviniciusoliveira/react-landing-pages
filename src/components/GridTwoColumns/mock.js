export default {
  title: 'Grid Two Title',
  sectionId: 'grid-two-columns',
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Saepe volupta',
  imageSrc: 'assets/images/javascript.svg',
};
