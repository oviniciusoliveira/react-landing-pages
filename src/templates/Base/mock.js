export default {
  links: [
    {
      children: 'Link 1',
      link: '#target1',
      newTab: false,
    },
    {
      children: 'Link 2',
      link: '#target2',
      newTab: false,
    },
    {
      children: 'Link 3',
      link: '#target3',
      newTab: false,
    },
    {
      children: 'Link 4',
      link: '#target4',
      newTab: false,
    },
    {
      children: 'Link 5',
      link: '#target5',
      newTab: false,
    },
    {
      children: 'Link 6',
      link: '#target6',
      newTab: false,
    },
    {
      children: 'Link 7',
      link: '#target7',
      newTab: false,
    },
    {
      children: 'Link 8',
      link: '#target8',
      newTab: false,
    },
    {
      children: 'Link 9',
      link: '#target9',
      newTab: false,
    },
    {
      children: 'Link 10',
      link: '#target10',
      newTab: false,
    },
  ],

  logoData: {
    text: 'Logo',
    link: '#',
  },
  footerHtml:
    '<p><a href="https://viniciusdeoliveira.net" target="_blank" >Feito com ❤ por Vinícius de Oliveira</a></p>',
};
