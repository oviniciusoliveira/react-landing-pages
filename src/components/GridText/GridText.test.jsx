import mock from './mock';
import { renderTheme } from '../../test/render-theme';
import { GridText } from '.';

describe('<GridText />', () => {
  test('should render', () => {
    const { container } = renderTheme(<GridText {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
