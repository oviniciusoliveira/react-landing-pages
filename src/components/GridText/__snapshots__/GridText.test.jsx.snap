// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridText /> should render 1`] = `
.c4 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c9 {
  color: #0A1128;
  font-size: 3.2rem;
  text-transform: none;
}

.c6 {
  font-size: 2.4rem;
}

.c2 .c5 {
  margin-bottom: 6.4rem;
}

.c7 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c8 .c3 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  left: 5rem;
}

.c8 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 7rem;
  top: -2.5rem;
  left: -5rem;
  -webkit-transform: rotate(5deg);
  -ms-transform: rotate(5deg);
  transform: rotate(5deg);
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (max-width:768px) {
  .c4 {
    font-size: 4.0rem;
  }
}

<div>
  <div
    class="c0"
    id="grid-text"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3 c4"
        >
          My grid
        </h2>
        <p
          class="c5 c6"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus malesuada arcu sed dictum malesuada.
        </p>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 1
            </h3>
            <p
              class="c5 c6"
            >
              Vivamus vitae porta diam. Nullam mauris lorem, pretium ut dapibus sed, efficitur ac lorem. Phasellus eget felis leo. Integer vel suscipit nulla. Integer eget gravida lorem. Fusce odio nibh, pellentesque nec ultricies eu, molestie non purus. Cras condimentum porta pharetra. Vivamus porta laoreet leo in iaculis. Donec eget laoreet risus. Etiam sed erat ultrices, sodales sem sit amet, sagittis lacus. Pellentesque in cursus enim. 
            </p>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 2
            </h3>
            <p
              class="c5 c6"
            >
              Vivamus vitae porta diam. Nullam mauris lorem, pretium ut dapibus sed, efficitur ac lorem. Phasellus eget felis leo. Integer vel suscipit nulla. Integer eget gravida lorem. Fusce odio nibh, pellentesque nec ultricies eu, molestie non purus. Cras condimentum porta pharetra. Vivamus porta laoreet leo in iaculis. Donec eget laoreet risus. Etiam sed erat ultrices, sodales sem sit amet, sagittis lacus. Pellentesque in cursus enim. 
            </p>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 3
            </h3>
            <p
              class="c5 c6"
            >
              Vivamus vitae porta diam. Nullam mauris lorem, pretium ut dapibus sed, efficitur ac lorem. Phasellus eget felis leo. Integer vel suscipit nulla. Integer eget gravida lorem. Fusce odio nibh, pellentesque nec ultricies eu, molestie non purus. Cras condimentum porta pharetra. Vivamus porta laoreet leo in iaculis. Donec eget laoreet risus. Etiam sed erat ultrices, sodales sem sit amet, sagittis lacus. Pellentesque in cursus enim. 
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
